include/zephyr/bluetooth/hci_types.h

```diff
@@ -3636,6 +3636,20 @@ struct bt_hci_evt_le_cs_subevent_result {
 	uint8_t steps[];
 } __packed;
 
+#define BT_HCI_EVT_LE_CS_SUBEVENT_RESULT_CONTINUE 0x32
+
+struct bt_hci_evt_le_cs_subevent_result_continue {
+	uint16_t conn_handle;
+	uint8_t config_id;
+	uint8_t procedure_done_status;
+	uint8_t subevent_done_status;
+	uint8_t procedure_abort_reason: 4;
+	uint8_t subevent_abort_reason: 4;
```

Comment:
This seems risky. I don't think it's safe to make assumptions of bitfield in-memory ordering, i.e. it's something that will depend on the architecture that this gets built for. I'd therefore suggest to use just a single `uint8_t` and provide helper macros to access the two sub-octet fields.

Reply:
It looks like there are multiple examples of bitfield access to protocol data in zephyr:

https://github.com/zephyrproject-rtos/zephyr/blob/52540d8412630edb2c64f4fcca1c1165bcce1ac3/subsys/bluetooth/controller/ll_sw/pdu.h#L417-L438

We have quite a few bitfields for the CS step data so this approach would be better for the user instead of access macros. So I suggest, we modify our bitfields similarly.

Reply:
@bugo-nordic @olivier-le-sage I just had to comment on the same thing here: https://github.com/zephyrproject-rtos/zephyr/pull/75637#discussion_r1800017520

Based on the text from the C standard, even though in practice the endianness-based variants may work, this is not something that's universally always guaranteed to be safe.

Reply:
I agree with you that the C standard doesn't define this, but the C standard also doesn't define that a byte is 8 bits or a pointer represents an address, or has size X on an X-bit CPU. It intentionally leaves these up to the ABI.

For instance if you search for bitfield or bit-field in the following documents:
* ARM EABI section 7.1.7 Bit-fields  (https://web.eecs.umich.edu/%7Eprabal/teaching/resources/eecs373/ARM-AAPCS-EABI-v2.08.pdf)
* i386 ABI section 3-6 (https://www.sco.com/developers/devspecs/abi386-4.pdf)
* x86-64 (https://raw.githubusercontent.com/wiki/hjl-tools/x86-psABI/x86-64-psABI-1.0.pdf)
* MIPS section 3.7 (https://refspecs.linuxfoundation.org/elf/mipsabi.pdf)
* RISCV section 2.1 (https://d3s.mff.cuni.cz/files/teaching/nswi200/202324/doc/riscv-abi.pdf)

I think you'll agree that any compiler for these targets would produce correct assembly from the Zephyr link layer code linked above.

Here is another example from the linux kernel: https://github.com/torvalds/linux/blob/master/include/uapi/linux/ip.h

Reply:
@olivier-le-sage fair enough, and thanks for the very good set of ABI references (which must have taken some time to collect!). I still think it's healthy to acknowldege the assumptions we're making, i.e. the correctness of the code is based on specific ABI definitions rather than the C standard. Personally, I tend to go for helper macros in this kind of situations, but I agree that from a code readability perspective it's nice to have bitfields in the structs. Anyway, since we have precedents in Zephyr as well as the Linux kernel, I'm ok with continuing with the endianness-based struct variant approach.

@gzh-terry based on the above, do you want to revise the approach you're taking in #75637?